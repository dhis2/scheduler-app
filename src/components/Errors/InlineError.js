import React from 'react'
import { string, arrayOf } from 'prop-types'
import { Title } from '../Title'
import styles from './InlineError.module.css'

const InlineError = ({ message, details }) => (
    <div className={styles.wrapper}>
        <Title priority={1}>Something went wrong</Title>
        <p>The error message was: &quot;{message}&quot;</p>
        {details.length > 0 && (
            <React.Fragment>
                <Title priority={2}>Details</Title>
                {details.map(detail => (
                    <p key={detail}>{detail}</p>
                ))}
            </React.Fragment>
        )}
    </div>
)

InlineError.defaultProps = {
    details: [],
}

InlineError.propTypes = {
    message: string.isRequired,
    details: arrayOf(string),
}

export default InlineError
