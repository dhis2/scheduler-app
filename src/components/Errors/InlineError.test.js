import React from 'react'
import { shallow } from 'enzyme'
import InlineError from './InlineError'

describe('<InlineError>', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <InlineError message="Error" details={['Details']} />
        )

        expect(wrapper).toMatchSnapshot()
    })
})
