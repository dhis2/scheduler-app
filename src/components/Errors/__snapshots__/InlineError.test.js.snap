// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineError> renders correctly 1`] = `
<div
  className="wrapper"
>
  <Title
    priority={1}
  >
    Something went wrong
  </Title>
  <p>
    The error message was: Error
  </p>
  <Title
    priority={2}
  >
    Details
  </Title>
  <p
    key="Details"
  >
    Details
  </p>
</div>
`;
